### ISSUE TITLE
The luminosity ratio of the focus indicator on the cells is less than 3:1 ratio under DataGridView

### ISSUE THREAD

**Issue by ayushanand-microsoft** [dotnet/winforms#9916]
### .NET version

.NET - 7.0.400
.NET Framework - 4.7.2

### Did it work in .NET Framework?

No

### Issue description

When the cell is selected inside a `DataGridView` and the focus is brought on the cell using `Tab`, then the luminosity ratio of the focus indicator on the cell is less than 3:1 ratio.

### Steps to reproduce

1. Create a Form with a DataGridView.
2. Add multiple columns to a DataGridView.
3. Add several rows to the DataGridView.
4. Select a row.
5. Use Tab to bring focus on a cell in the selected row.

**Actual behavior:**
Focus indicator border line of the DataGridView control has very low luminosity contrast ratio.

![image](https://github.com/dotnet/winforms/assets/144995455/f74130b9-df4a-4231-a395-f667ba557611)

**Expected behavior:**
Luminosity contrast ratio of the focus indicator border line of the DataGridView control should be greater than 3:1.

**Comment by merriemcgaw** [comment]
@Olina-Zhang can you confirm behavior on vanilla .NET 8 and .NET 4.8.1?

**Comment by merriemcgaw** [comment]
@ayushanand-microsoft do you have a simple sample app targeting .NET 7 or 8 that demonstrates this issue? The application you have above looks like it's .NET Framework and this repo deals only with .NET.

**Comment by Tanya-Solyanik** [comment]
@Olina-Zhang - could you please verify this issue in our accessibility test app on .net7 and .net8

**Comment by merriemcgaw** [comment]
@Olina-Zhang could you also check and see if you can figure out a workaround for the scenario when the whole row is selected and we want to show the which cell is in focus at the time.

**Comment by Olina-Zhang** [comment]
@Tanya-Solyanik @merriemcgaw Verified it in .Net 7.0 and .Net 8.0, they have the same result:
**.Net 8.0:**
![image](https://github.com/dotnet/winforms/assets/26474449/72f1ee40-bac4-4b5f-b6f5-eecfc32ce073)

**Workaround:** Custom DataGridViewCellStyle's SelctionBackColor in order to the luminosity ratio of the focus indicator on the cells passed and the luminosity ratio of cell text and selection backColor passed.
For example: 
add this code: dataGridViewCellStyle1.SelectionBackColor = Color.DarkViolet;

**The result of luminosity ratio of the focus indicator on the cells:** --- passed
![image](https://github.com/dotnet/winforms/assets/26474449/be9fd149-333d-4fb7-90fc-c362a6e4f87c)

**The result of luminosity ratio of cell text and selection backColor** --- passed
![image](https://github.com/dotnet/winforms/assets/26474449/6d95b1d4-78cb-4f1d-8406-da2d5fafdc2d)

**Comment by Zheng-Li01** [comment]
@ricardobossan verified the issue with latest .NET 9.0.100-alpha.1.23563.6 on win10 & win11 OS, there have the same results between **Contrast between background and focus indicator** and **Contrast between background and text** that the luminosity ratio is 4.498:1, and there have a fail about "Regular text" as below screenshot, want to confirm with you if the current results is expected?
![image](https://github.com/dotnet/winforms/assets/38325459/2b992bb7-965c-4db6-be9d-3d49c5eb2d76)

**Comment by ricardobossan** [comment]
@Zheng-Li01 The issue was about the contrast ratio being less then 3:1, which has been fixed.

Even so, I tried now changing the `DataGridViewTextBoxCell`'s background color from `cellStyle.ForeColor` to others, until I found `System.HotTrack`, which was very similar. It solved the fail at Regular text, and kept working on all contrast themes, except `Night Sky`, in which the `Regular text` problem happened as it did with the original `cellStyle.ForeColor` when not in the high contrast themes (at a lower Ratio than before though: 4.348:1). Bellow are screenshots:

## Code Change

![image](https://github.com/dotnet/winforms/assets/30190295/cc950541-6f70-477e-8ac1-689a911f78f9)

## Without High Contrast Theme

![image](https://github.com/dotnet/winforms/assets/30190295/ff6cc70b-5171-4e7c-bb56-c26682fff667)

## Aquatic Theme

![image](https://github.com/dotnet/winforms/assets/30190295/fddee48c-7a9c-420f-bd32-a8df091deff5)

## Desert Theme

![image](https://github.com/dotnet/winforms/assets/30190295/a10fabdc-fe7b-475e-8a3a-00865749dea3)

## Dusk Theme

![image](https://github.com/dotnet/winforms/assets/30190295/88962685-bcd8-418c-bc62-a01f11adac6d)

## Night sky Theme

![image](https://github.com/dotnet/winforms/assets/30190295/b8e4cc9b-0c8e-4fec-9ddc-1ae7e81db661)

@Tanya-Solyanik Should we consider that the issue was solved as it has a ratio higher than 3:1? Should we change the background color to `System.HotTrack`? How should we proceed?

**Comment by Tanya-Solyanik** [comment]
@ricardobossan @Zheng-Li01  - Contrast ratio requirements for informative text and for UI elements, like focus borders, are different, this specific bug was about the border contrast only, and this part is solved.

Zheng - good point about the text not being contrasting enough, but the ratio is close to the required 4.5, and we had gotten exceptions for cases like this one in the past, so I think this won't be a problem to get an exception again if we have customer complaints. So this issue is a "won't fix" for now.
